### PR TITLE
Update DNS version in kubeadm of 1.10 cycle.

### LINKED ISSUE #622: Bump minimum version to v1.9 and cleanup v1.8-specific code at HEAD
TODO:
 - Bump minimum version: https://github.com/kubernetes/kubernetes/pull/57233, done!
 - kube-proxy cleanup: https://github.com/kubernetes/kubernetes/pull/57305, done!
 - kube-dns cleanup: https://github.com/kubernetes/kubernetes/pull/57271, done!
 - Cleanup old upgrading code that is v1.8->v1.9-specific

ref: https://github.com/kubernetes/kubeadm/blob/master/docs/release-cycle.md

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
Since kubeadm in 1.10 will not support 1.8 kubernetes, update DNS version accordingly.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kubeadm/issues/622

**Special notes for your reviewer**:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews 

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
pull-kubernetes-bazel-test pending #57230

File: cmd/kubeadm/app/phases/addons/dns/dns.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func kubeDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interfac
 			DNSBindAddr:    dnsBindAddr,
 			DNSProbeAddr:   dnsProbeAddr,
 			DNSDomain:      cfg.Networking.DNSDomain,
-			DNSProbeType:   GetKubeDNSProbeType(k8sVersion),
```
Comment (luxas) [inline cmd/kubeadm/app/phases/addons/dns/dns.go]: I'd just build this into the manifest again; and not have it at all as a gotmpl parameter

**Review by luxas — COMMENTED** [review]
One nit, otherwise looks good

**Comment by xiangpengzhao** [comment]
@luxas nit addressed. PTAL. Thanks!

**Review by luxas — APPROVED** [review]
/approve no-issue
/kind cleanup

@xiangpengzhao needs a rebase and then we're good to go

**Comment by xiangpengzhao** [comment]
@luxas rebased. But this PR should wait for #57230.

**Review by luxas — APPROVED** [review]
/lgtm

@xiangpengzhao as the etcd PR is on hold, can you make this mergeable first by fixing the tests?

**Comment by xiangpengzhao** [comment]
/hold
will fix later at the weekend :)

File: cmd/kubeadm/app/constants/constants.go [hunk 2]
```diff
@@ -236,6 +236,7 @@ var (
 		8:  "3.0.17",
 		9:  "3.1.10",
 		10: "3.1.10",
+		11: "3.1.10",
```
Comment (xiangpengzhao) [inline cmd/kubeadm/app/constants/constants.go]: This line is just to make the test pass. Should bump to 3.2.12 done in #57230

**Review by luxas — APPROVED** [review]
/lgtm

Thanks for making this one mergeable :+1:!

**Pull request merged**